### DISCUSSION TITLE
Still no plans for MAC OSX Support??

### DISCUSSION THREAD

**Discussion by kiwivideo** [SuRGeoNix/Flyleaf#478] [Ideas]
If still no LINUX, any chance of MacOS version?  Runs brilliantly on Mac under Parallels, but need native Mac Media Player with frame-by-frame BACKWARDS that VLC etc can't do. I had hoped Flyleaf might be a quick way to implement FFMpeg on MacOS... Or am I missing something??  Thanks!

**Comment by SuRGeoNix** [comment]
Hi @kiwivideo, that's the plan with v4 that I'm currently struggling :) To support all the FFmpeg's hw accelerators and then use multiple renderer's / control's as well. I've finished the low level API for mux/demux/encode/decode/filter (custom FFmpeg.Autogen that will simplify the FFmpeg's API to OOP and even separated by AVS / DE (audio-video-subtitle / decoding-encoding). It already supports 'all' OSes and hw accels. Now working on threading before jumping to rendering / controls / playback.

**Reply by kiwivideo** [reply]
WOW! Fantastic.  Sounds like you maybe have the harder bits done ??!  Very impressed with what you've achieved.  Starts to replicate what I built myself many years ago on old Windows technology, but need to upgrade to MAUI now.  Please keep us in the loop as it develops. Thanks.

**Reply by SuRGeoNix** [reply]
Well, the most difficult part is to 'read' (research that you can't really do with documentation) FFmpeg's code and indeed that part is done now. Now a good design to mix higher level 'plugins' , 'filters' etc. to match FFmpeg's logic, by using efficient threading / queues should be the key for an (at least) sufficient library ;)

**Reply by kiwivideo** [reply]
Great!  You seem to be on a very similar path in methodology and design requirements that my team and I engaged in decades ago at the birth of the digital revolution.  When solutions to the identified NEEDS don't yet exist, it's not good enough to build "from the ground up".  You have to go "below ground". It appears your "mining" efforts are paying off.  A long, slow, and potentially costly process.  I hope you're getting enough rewards along the way to keep you going.  MUCH APPRECIATED!

**Comment by umlx5h** [comment]
In terms of cross-platform support, is the GUI planned to support both Avalonia and MAUI? Or one or the other?
